### ISSUE TITLE
Python 3.7 logging occasionally hangs on long running jobs

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#579]
I've noticed a few times that Python 3.7 has hung on very long running jobs (experiments, ~24 hours) that previously had no such issues.  Using `strace` (well techincally `dtruss`, but similar idea) it looked to me like a threadlock.  I googled a bit and it seems there was in fact a collection of problems in Python 3.7 using the standard library logging with either threading or multiprocessing (we use both in different places).  Moving to 3.8 resolved it for me (never saw in 3.6 either)

Mainly this issue is for awareness. I don't think there is much to do about it now, and I'm not sure 3.7 is actually getting updates anymore anyway. As 3.11 comes online this should go away naturally by dropping 3.7.

Bottom line, don't use 3.7 for production or very long running jobs. It should be fine otherwise.

**Comment by garrettwrong** [comment]
This hasn't come up again, believe it was resolved.  Closing.
